### DISCUSSION TITLE
Possible to recursively / automatically add header to old projects?

### DISCUSSION THREAD

**Discussion by nextfour-back-sabina** [davidquinn/psi-header#40] [Q&A] [answered]
Hi and thank you for providing a nice extension!

I am wondering if it is possible to add license-headers to all files in an existing project without manually making changes in files?

**Comment by davidquinn** [comment, marked as answer]
Hi, and glad you like the extension.

I am sorry, but this is not possible for the reasons discussed in [this discussion](https://github.com/davidquinn/psi-header/discussions/2#discussion-3240114).

In a nutshell, the problem is that the extension can only update files that are (a) open in the VSC editor; and (b) VSC reports that the file has been modified.

Regards
David

**Reply by nextfour-back-sabina** [reply]
Thank you for replying so fast! Too bad it's not possible, but I created a script for it so at least my problem solved.

**Reply by davidquinn** [reply]
The extension can check if an existing file has a header and add it if it doesn't find one during save. I tend to rely on that when using old projects so the files get headers as and when I need to edit them.

But your script idea works as well and the extension will continue to maintain the header so long as it inserts the header in the same format as the extension is configured for.

D

**Reply by leonardbiofi** [reply]
> Thank you for replying so fast! Too bad it's not possible, but I created a script for it so at least my problem solved.

Could you please provide your script to do this ? 

I am also interested in such script
